wire-runtime-swift/src/main/swift/ProtoExtensible.swift

```diff
@@ -19,3 +19,29 @@ import Foundation
 public protocol ProtoExtensible {
     var unknownFields: UnknownFields { get set }
 }
+
+public extension ProtoExtensible {
+    func parseUnknownField<T: ProtoDecodable>(
+        with protoDecoder: ProtoDecoder = .init(),
+        fieldNumber: UInt32,
+        type: T.Type
+    ) -> T? {
+        guard let data = unknownFields[fieldNumber] else {
```

Comment:
Realistically, we should have a second cache for extensible types which avoids constantly encoding/decoding the raw bytes

eg `if let cached = cache[fieldNumber] as? T { return cached }`

Reply:
yes, @efirestone also brought up memoization here. Note implementation is not nearly close to complete here.

Reply:
- [ ] TODO

Reply:
this is pushed. I also did performance testing and did not notice any difference before and after.